zebra-consensus/src/primitives/halo2/tests.rs

```diff
@@ -0,0 +1,146 @@
+                Anchor::from_bytes(anchor_bytes).unwrap(),
+            );
+
+            for _ in 0..num_recipients {
+                builder
+                    .add_recipient(None, recipient, NoteValue::from_raw(note_value), None)
+                    .unwrap();
+            }
+
+            let bundle: Bundle<_, i64> = builder.build(rng).unwrap();
+
+            let bundle = bundle
+                .create_proof(&PROVING_KEY)
+                .unwrap()
+                .apply_signatures(rng, [0; 32], &[])
+                .unwrap();
+
+            zebra_chain::orchard::ShieldedData {
+                flags,
+                value_balance: note_value.try_into().unwrap(),
+                shared_anchor: anchor_bytes.try_into().unwrap(),
+                proof: zebra_chain::primitives::Halo2Proof(
+                    bundle.authorization().proof().as_ref().into(),
+                ),
+                actions: bundle
+                    .actions()
+                    .iter()
+                    .map(|a| {
+                        let action = zebra_chain::orchard::Action {
+                            cv: a.cv_net().to_bytes().try_into().unwrap(),
+                            nullifier: a.nullifier().to_bytes().try_into().unwrap(),
+                            rk: <[u8; 32]>::from(a.rk()).try_into().unwrap(),
+                            cm_x: pallas::Base::from_bytes(&a.cmx().into()).unwrap(),
+                            ephemeral_key: a.encrypted_note().epk_bytes.try_into().unwrap(),
+                            enc_ciphertext: a.encrypted_note().enc_ciphertext.into(),
+                            out_ciphertext: a.encrypted_note().out_ciphertext.into(),
+                        };
+                        zebra_chain::orchard::shielded_data::AuthorizedAction {
+                            action,
+                            spend_auth_sig: <[u8; 64]>::from(a.authorization()).into(),
+                        }
+                    })
+                    .collect::<Vec<_>>()
+                    .try_into()
+                    .unwrap(),
+                binding_sig: <[u8; 64]>::from(bundle.authorization().binding_signature())
+                    .try_into()
+                    .unwrap(),
+            }
+        })
+        .collect();
+
+    // Use separate verifier so shared batch tasks aren't killed when the test ends (#2390)
+    let mut verifier = Fallback::new(
+        Batch::new(
+            Verifier::new(&VERIFYING_KEY),
+            crate::primitives::MAX_BATCH_SIZE,
+            crate::primitives::MAX_BATCH_LATENCY,
+        ),
+        tower::service_fn(
+            (|item: Item| ready(item.verify_single(&VERIFYING_KEY).map_err(Halo2Error::from)))
+                as fn(_) -> _,
+        ),
+    );
+
+    // This should fail if any of the proofs fail to validate.
+    verify_orchard_halo2_proofs(&mut verifier, shielded_data)
+        .await
+        .unwrap()
```

Comment:
Suggestion: I usually use at least one assertion to mark what the test is focused on checking.

```suggestion
    assert!(verify_orchard_halo2_proofs(&mut verifier, shielded_data)
        .await
        .is_ok());
```